backend/src/main/kotlin/com/ksidelta/libruch/modules/organisation/OrganisationController.kt

```diff
@@ -0,0 +1,67 @@
+package com.ksidelta.libruch.modules.organisation
+
+import com.ksidelta.libruch.modules.kernel.Party
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.future.await
+import kotlinx.coroutines.withTimeout
+import org.axonframework.commandhandling.gateway.CommandGateway
+import org.axonframework.eventhandling.EventBus
+import org.axonframework.messaging.MetaData
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import java.util.*
+import kotlin.reflect.KClass
+
+@RestController
+@RequestMapping("/api/organisation")
+class OrganisationController(val commandGateway: CommandGateway, val eventBus: EventBus) {
+
+    @PostMapping
+    suspend fun create(user: Party.User, @RequestBody createOrganisationDTO: CreateOrganisationDTO) {
+        eventBus.awaitingEvent(UserToOrganisationsModelViewUpdated::class) { correlationId ->
+            createOrganisationDTO.let {
+                commandGateway.send<Unit>(
+                    CreateOrganisation(it.name, user),
+                    MetaData(mutableMapOf(Pair("traceId", correlationId)))
+                )
+            }.await()
+        }
+
+
+    }
+}
+
+suspend fun EventBus.awaitingEvent(eventType: KClass<*>, block: suspend (correlationId: String) -> Unit) {
+    withTimeout(10000) {
+        val correlationId = UUID.randomUUID().toString()
+
+        val flow =
+            callbackFlow {
+                val x = this
+                val subscription = this@awaitingEvent.subscribe { events ->
```

Comment:
Here we wait for our lovely event.
There should be separate shared EventHandler which should make it more optimized for searching via type and correlationId.
I think that some simple Map would be enough.

It would be nice if axon had some utility service for it.